src/eq_schema/RoutingConditions.js

```diff
@@ -8,33 +10,53 @@ const getAllOptions = condition => {
   }
 };
 
-const createOptionByIdLookup = flow(
+const keyOptionsById = flow(
   flatMap(getAllOptions),
   keyBy("id")
 );
 
+const createBuilder = (answerType, condition, conditions) => {
+  let builder = null;
+
+  switch (answerType) {
+    case "Radio":
+      builder = new RadioRoutingCondition(
+        condition,
+        keyOptionsById(conditions),
```

Comment:
AFAICT this is an object of all conditions by id and the below is a list of all condition ids.

If that's the case then the second is just `Object.keys` of the first?

Reply:
Not quite, `keyOptionsById` finds all Options within all the routing conditions  for the routing rule and keys the Options by their ids for easier lookup.

Object.keys will not work in this case because the third argument is a list of condition Ids (not the option Ids).

It's probably worth noting that I haven't changed any behaviour here for routing based on radio options. I've just pulled the behaviour out into its own class and refactored the argument passing. So it may be necessary to revisit how we do this.

The original implementation was lacking tests so I wrote a few tests to try and document existing behaviour and to ensure that I didn't break anything during refactoring. Routing on radio options wasn't the main focus of this task.